File: Practica 4/CivitasJava4/CivitasJava/src/civitasjava/CasillaJuez.java
package civitasjava;

import java.util.ArrayList;

public class CasillaJuez extends Casilla {
        
    private int carcel; //Solo tendremos una casilla juez
    
    // Construye casilla de tipo JUEZ
    CasillaJuez(int numCasillaCarcel, String nombre){ // Solo debe construirlas el metodo añadeJuez
        super(nombre);
        this.init();
        
        this.carcel = numCasillaCarcel;
        setCarcel(numCasillaCarcel); 
    }   
    
    private void init(){
        carcel = 0;           
    }     
    
    /**
     * @brief Encarcela a el jugador dado(si existe)
     * @param actual, el indice del jugador
     * @param todos, el vector de jugador
     * @return void
     */
    @Override
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if( jugadorCorrecto(iactual, todos) ){
            super.recibeJugador(iactual, todos);
            
            todos.get(iactual).encarcelar(carcel);
        }
    }
}

File: Practica 4/CivitasRuby4/CivitasRuby/lib/casilla_juez.rb
module Civitas
  
class CasillaJuez < Casilla
  
# @brief Método que hace de Constructor de la clase. Casilla juez
# @param carcel, El número de la casilla cácel
# @param nombre, el nombre de la casilla
# @return Un objeto de clase casilla
  def initialize(carcel, nombre)
    super(nombre)
    Casilla.carcel=carcel
    @@carcel = carcel
  end

# @brief Encarcela a el jugador dado(si existe)
# @param actual, el indice del jugador
# @param todos, el vector de jugador
# @return Es void
public
def recibe_jugador(actual,todos)
  if(jugador_correcto(actual,todos))
    super(actual,todos)
    todos[actual].encarcelar(@@carcel)
  end
end

end
end